File: soon/MemoryAnalyzer/src/network/Protocol.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package network;

/**
 *
 * @author master
 */
public class Protocol {    
    // Constant define a timeouts
    public static final int CONNECTION_TIMEOUT = 10;
    
    // Symbol delimiters of protocol
    public final static String COM_DELIMITER = "`";
    public final static String ARGS_DELIMITER = "~";
    
    // Answers of protocol
    public final static String SUCCESS = "suc";
    public final static String ERROR = "err";
    
    // Commands of protocol
    public final static String HI = "hi";
    public final static String BYE = "bye";
    public final static String PIN_INIT = "p_in";
    public final static String PIN_EXEC = "p_ex";
    public final static String GET_BINARY = "gbin";
}

File: PinServer/protocol.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "protocol.hpp"

bool isBigEndian() {
    int value = 0;
    ((char*)&value)[0] = 1;
    return value != 1;
}
bool isLittleEndian() {
    int value = 0;
    ((char*)&value)[0] = 1;
    return value == 1;
}
void hton(byte* arr, int size) {
    if(isBigEndian()) {
        return;
    }
    byte tmp;
    int half_size = size / 2;
    for(int i = 0; i < half_size; i++) {
        tmp = arr[i];
        arr[i] = arr[size - i - 1];
        arr[size - i - 1] = tmp;
    }
}
void ntoh(byte* arr, int size) {
    if(isBigEndian()) {
        return;
    }
    byte tmp;
    int half_size = size / 2;
    for(int i = 0; i < half_size; i++) {
        tmp = arr[i];
        arr[i] = arr[size - i - 1];
        arr[size - i - 1] = tmp;
    }
}